### ISSUE TITLE
Additional Exercises?

### ISSUE THREAD

**Issue by dougflip** [alpacaaa/zero-bullshit-haskell#11]
Hello @alpacaaa! First off, thanks for this repo - I've had a great time working through the exercises! I've gone through them once by myself and I'm taking some people at work through who have zero prior Haskell experience. Its been great!

I was curious if you have plans to extend it beyond 8 exercises? If so, is there anything I can do to help? I'm still somewhat of a novice Haskell user but would love to help if possible! Thanks again!

**Comment by alpacaaa** [comment]
Hey @dougflip! You're right, I should be adding more content :)

Where do you think should the exercises go from here? As a novice, do you think exercises have covered enough for you to feel somewhat productive with the language?

To me, the next big thing would be introducing `IO`. Have exercises that make HTTP requests or maybe save and read to files.

Thanks for the feedback!

**Comment by dougflip** [comment]
I was thinking `IO` would be a neat next step as well. I saw that you already have an `effectfulHandler` as part of the API. Maybe we could read/write your shopping cart to a file instead of a just keeping it in memory? The JS tests would be largely the same I guess - but the backend implementation would obviously be different.

I hadn't though about making HTTP requests, but that would be really cool too! Did you have any ideas about what service(s) could be used?

**Comment by alpacaaa** [comment]
Yes, `effectfulHandler` is already there just waiting to be used! :D
Your suggestion of keeping exercises as they are but introduce state is a neat idea. I was thinking of working with random REST apis (fetch posts from Reddit or issues/commits from a repo on Github).

I'll get to it eventually, thanks for the feedback and let me know if you have other suggestions :)

**Comment by MatthewZito** [comment]
@alpacaaa Hi, wonderful series - thank you for making this. It's very difficult to find project-driven or otherwise interactive learning resources for Haskell; this series is an absolute gem!

I hope to see more exercises, too. I think @dougflip makes a good suggestion in expanding existing material to include more concepts e.g. IO, state, maybe a simple data service layer with a key/value store?

I'd be happy to help (I'm obviously very novice at Haskell, but I can write JS ;-) ). Thanks!

**Comment by alpacaaa** [comment]
Hi @MatthewZito! I'd love to come back to this project eventually, I'm busy with other stuff and can't give it the attention it deserves. If you have concrete ideas of exercises/small projects you think it'd be interesting to build please post them here! Even if they're written in JS that's perfectly fine :) I'm always on the lookout for more beginner level material to teach Haskell, so ideas welcome.

**Comment by dougflip** [comment]
@MatthewZito just to add something else to the mix - I picked up a copy of this yesterday: https://leanpub.com/simple-haskell-book. You sound like a similar background as me (working in JS but learning Haskell) so you may be interested too :)

@alpacaaa thanks again for your work focusing on simple and practical Haskell. It really is helpful for those of us trying to pick up Haskell on the side!
